src/parquet/bloom.h

```diff
@@ -18,174 +18,140 @@
 #ifndef PARQUET_BLOOM_H
 #define PARQUET_BLOOM_H
 
-#include <set>
 #include <cstdint>
 
+#include "parquet/hasher.h"
 #include "parquet/types.h"
 #include "parquet/util/memory.h"
 
 namespace parquet {
 class OutputStream;
 
-// Bloom Filter is a compact structure to indicate whether an item is not in set or
-// probably in set. Bloom class is underlying class of Bloom Filter which stores a
-// bit set represents elements set, hash strategy and bloom filter algorithm.
+// A Bloom filter is a compact structure to indicate whether an item is not in set or
+// probably in set. Bloom class is underlying class of Bloom filter which stores a
+// bit set represents elements set, hash strategy and Bloom filter algorithm.
 
-// Bloom Filter algorithm is implemented using block Bloom filters from Putze et al.'s
-// "Cache-,Hash- and Space-Efficient Bloom Filters". The basic idea is to hash the
-// item to a tiny Bloom Filter which size fit a single cache line or smaller. This
-// implementation sets 8 bits in each tiny Bloom Filter. Tiny bloom filter are 32
+// Bloom filter algorithm is implemented using block Bloom filters from Putze et al.'s
+// "Cache-,Hash- and Space-Efficient Bloom filters". The basic idea is to hash the
+// item to a tiny Bloom filter which size fit a single cache line or smaller. This
+// implementation sets 8 bits in each tiny Bloom filter. Tiny Bloom filter are 32
 // bytes to take advantage of 32-bytes SIMD instruction.
 
 class Bloom final {
  public:
-  // Hash strategy available for bloom filter.
-  enum HashStrategy {
-    MURMUR3_X64_128
-  };
+  // Hash strategy available for Bloom filter.
+  enum class HashStrategy { MURMUR3_X64_128 = 0 };
 
   // Bloom filter algorithm.
-  enum Algorithm {
-    BLOCK
-  };
+  enum class Algorithm { BLOCK = 0 };
 
-  /**
-   * Default false positive probability value use to calculate optimal number of bits
-   * used by bloom filter.
-   */
-  static constexpr double DEFAULT_FPP = 0.01;
-
-  // Bloom filter data header, including number of bytes, hash strategy and algorithm.
-  static constexpr int HEADER_SIZE = 12;
-
-  // Bytes in a tiny bloom filter block.
+  // Bytes in a tiny Bloom filter block.
   static constexpr int BYTES_PER_FILTER_BLOCK = 32;
 
-  // Default seed for hash function, to keep cross compatibility the seed is same as
-  // value defined in parquet-mr.
-  static constexpr int DEFAULT_SEED = 104729;
-
-  // Default maximum bloom filter size (need to discuss)
+  // Default maximum Bloom filter size (need to discuss)
   static constexpr int DEFAULT_MAXIMUM_BLOOM_FILTER_BYTES = 16 * 1024 * 1024;
```

Comment:
OK. Can we leave it is for upper bound check and update comment only?

Reply:
It's pretty low for a check. Maybe increase the size to something really enormous like 1TB and rename it MAX_SANE_FILTER_SIZE?

Reply:
Since old lower boundary depends on constant BYTES_PER_FILTER_BLOCK which will be move out of Bloom class for algorithm abstract, So I 'm going to add two macros named MINIMUM_BLOOM_FILTER_BYTES and MAXIMUM_BLOOM_FILTER_BYTES to replace old boundaries. MINIMUM_BLOOM_FILTER_BYTES would be 64 bytes as same as cache alignment, and MAXIMUM_BLOOM_FILTER_BYTES would be 256MB (HADOOP default block size), how about this?